neuralforecast/models/timellm.py

```diff
@@ -395,29 +401,122 @@ def __init__(
         self.enc_in = enc_in
         self.dec_in = dec_in
 
-        self.llm_config = llm_config
-        self.llm = llm
-        self.llm_tokenizer = llm_tokenizer
+        self.llm_model = llm_model
+        self.llm_num_hidden_layers = llm_num_hidden_layers
+        self.llm_output_attention = llm_output_attention
+        self.llm_output_hidden_states = llm_output_hidden_states
 
-        if self.llm is None:
-            if not IS_TRANSFORMERS_INSTALLED:
-                raise ImportError(
-                    "Please install `transformers` to use the default LLM"
+        if self.llm_model == "LLAMA":
```

Comment:
Essentially this limits users to three possible LLMs - isn't there an option where we keep the try/except construct but are able to feed it any LLM name?

Reply:
I’ve updated it to use AutoModel, AutoTokenizer, and AutoConfig, so you can now use pretrained model paths beyond the three specific models.